src/App.tsx

```diff
@@ -1,24 +1,311 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useState, useMemo,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Footer } from './components/footer';
+import { Main } from './components/main';
+import { ErrorNotification } from './components/errorNotification';
+import { Header } from './components/header';
+import { Todo } from './types/Todo';
+import { SelectOptions } from './types/SelectOptions';
+import { ErrorType } from './types/ErrorType';
+import {
+  deleteTodos, getTodos, postTodos, patchTodoStatus, patchTodoTitle,
+} from './api/todos';
+import { USER_ID } from './utils/userId';
+import { filterTodosBySelectOptions } from './utils/filterTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorType, setErrorType] = useState<ErrorType>(ErrorType.ADD);
+  const [completedTodosId, setCompletedTodosId] = useState<number[]>([]);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isResponce, setIsResponce] = useState(true);
+  const [todoLoadingId, setTodoLoadingId] = useState<number[]>([]);
+  const [selectedOption, setSelectedOption]
+    = useState<SelectOptions>(SelectOptions.ALL);
+
+  const filteredCompletedTodosId = useCallback((allTodos: Todo[]) => () => {
```

Comment:
Get used to name your functions according to these rules
![image](https://user-images.githubusercontent.com/96409563/224943132-86353978-7ddf-4bc2-a20a-0e01fb87b288.png)